neurom/features/neurite_features.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2015, Ecole Polytechnique Federale de Lausanne, Blue Brain Project
+# All rights reserved.
+#
+# This file is part of NeuroM <https://github.com/BlueBrain/NeuroM>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+''' Neurite Related Features'''
+
+from neurom.core.types import TreeType
+from neurom.core.types import tree_type_checker as _ttc
+from neurom import segments as _seg
+from neurom import sections as _sec
+from neurom import bifurcations as _bifs
+from neurom import iter_neurites
+
+
+def section_lengths(neurites, neurite_type=TreeType.all):
```

Comment:
I am looking at the amount of code that has to be written for  a single feature, and it looks like too much. That leads me to think that the enum is a bad idea (sorry) and that we might as well have functions. Now, in most cases, these functions map to something without parameters other than a tree type filter (e.g. `get_segment_length`). So we have a most common use case where we have a functor (e.g. `sections.length`), `iter_neurites`, a tree type, and we want to return a `numpy` array. I think this can be easily expressed in terms of a decorator and the already existing functions. Then we only really need to write code for the functions that require extra parameters.